### PR TITLE
#83: Add bounded merge-loop command

### LINKED ISSUE #83: Add bounded merge-loop command
## Issue Setup

- UAT Required: No
- Related Parent Issue or Context: Dogfood readiness gap for continuous merge polling

## Issue Goal

Add a bounded `merge-loop` CLI command that repeatedly invokes the guarded merge lane for Project issues in `Merging`.

## Why Now

Jade Symphony can currently land one `Merging` issue with `merge-once`, but the operator still has to run that command manually for each item. The dogfood readiness docs explicitly list long-running `merge-loop` polling as missing. A bounded loop is the next safe step toward self-running merge-lane operation.

## Issue Context

`merge-once` already centralizes the guarded land flow: it selects one `Merging` issue, runs PR preflight, writes workpad evidence, merges clean PRs with explicit `--write`, or routes blockers. The new command should reuse that primitive instead of duplicating merge-lane logic.

### Decisions

- `merge-loop` should mirror the safety posture of `run-loop` and `review-loop`.
- `--once` should be equivalent to `--max-iterations 1`.

### Assumptions

- A bounded loop is enough for this slice; unbounded idle polling can remain a follow-up if it needs additional operator controls.

## Non-Negotiable Guardrails

- Do not bypass `merge-once` safety checks.
- Do not set `Human Review`.
- Do not change Agent Review behavior.
- Do not edit files under `docs/bootstrap/references/openai-symphony`.

### In Scope

- Add CLI command `merge-loop <workflow>`.
- Support `--max-iterations <n>`, `--once`, `--dry-run`, and `--write`.
- Reuse the existing `merge-once` primitive for each tick.
- Stop after the bounded iteration count or when no `Merging` issue remains.
- Add parser and loop behavior tests.
- Update README and dogfood readiness docs.

### Out of Scope

- Unbounded idle polling.
- Worker supervision.
- Changing merge lane decisions beyond the existing guarded primitive.

### Target Repository / Package

- `Alive24/jade-symphony`

### Relevant Knowledge Sources

- `docs/bootstrap/JADE_WORKFLOW.md`
- `docs/bootstrap/JADE_SYMPHONY_SPEC.md`
- `README.md`
- `docs/dogfood-readiness.md`

### Relevant Code Paths

- `src/main.rs`
- `src/merge_lane.rs`

## Current State

`merge-once` exists and is documented, but there is no `merge-loop` command. Dogfood readiness still lists `merge-loop` as missing.

## Deliverable Shape

A CLI command, tests, and docs that provide a bounded merge-lane loop without weakening the existing merge safety model.

## Risks or Constraints

- The loop must not accidentally spin forever in write mode.
- The loop must keep `--write` explicit for mutation.
- The loop must not hide individual `merge-once` outcomes.

## Expected Outcome

Operators can run a bounded merge pass with a command such as `cargo run -- merge-loop examples/github-project-workflow.md --max-iterations 3 --write`.

### Completion Criteria

- Parser tests cover `merge-loop`, `--once`, `--max-iterations`, `--dry-run`, and `--write`.
- Loop tests cover bounded iteration behavior.
- README or dogfood readiness no longer claims `merge-loop` is missing for the bounded case.

### Functional Verification

- `cargo test`
- `cargo fmt --check`
- `cargo clippy --all-targets --all-features -- -D warnings`

### UAT

- Not required.

### Context Verification

- Confirm the command still matches the bootstrap rule that `Merging` is the land-flow state.

### PR DESCRIPTION
**Pull request by Alive24**
## Summary

- Add bounded `merge-loop` CLI command that reuses guarded `merge-once` ticks.
- Support `--max-iterations`, `--once`, `--dry-run`, and `--write`.
- Stop when no `Merging` issue remains or when the explicit iteration limit is reached.
- Update README and dogfood readiness docs.

## Verification

- `cargo test`
- `cargo fmt --check`
- `cargo clippy --all-targets --all-features -- -D warnings`
- `cargo run -- merge-loop examples/github-project-workflow.md --max-iterations 1 --dry-run`

Closes #83

### CODE REVIEW

**Review by Alive24 — COMMENTED** [review]
## 📋 Review Summary
Skipped due to error: Command '['gemini', '-p', '\nYou are a Code Review Architect. Follow these guidelines strictly:\n---\nname: code-review-commons\ndescription: Common guidelines, persona and critical constraints for performing high-quality code reviews. Use this skill when performing a /code-review or /pr-code-review command.\nuser-invokable: false\n---\n\n# Code Review Commons\n\n## PERSONA\n\nYou are a very experienced **Principal Software Engineer** and a meticulous **Code Review Architect**. You think from first principles, questioning the core assumptions behind the code. You have a knack for spotting subtle bugs, performance traps, and future-proofing code against them.\n\n## OBJECTIVE\n\nYour task is to deeply understand the **intent and context** of the provided code changes (diff content) and then perform a **thorough, actionable, and objective** review.\nYour primary goal is to **identify potential bugs, security vulnerabilities, performance bottlenecks, and clarity issues**.\nProvide **insightful feedback** and **concrete, ready-to-use code suggestions** to maintain high code quality and best practices. Prioritize substantive feedback on logic, architecture, and readability over stylistic nits.\n\n## Instructions\n\n1. **Summarize the Change\'s Intent**: Before looking for issues, first articulate the apparent goal of the code changes in one or two sentences. Use this understanding to frame your review.\n2. **Establish context** by reading relevant files. Prioritize:\n    a. All files present in the diff.\n    b. Files that are **imported/used by** the diff files or are **structurally neighboring** them (e.g., related configuration or test files).\n3. **Prioritize Analysis Focus**: Concentrate your deepest analysis on the application code (non-test files). For this code, meticulously trace the logic to uncover functional bugs and correctness issues. Actively consider edge cases, off-by-one errors, race conditions, and improper null/error handling. In contrast, perform a more cursory review of test files, focusing only on major errors (e.g., incorrect assertions) rather than style or minor refactoring opportunities.\n4. **Analyze the code for issues**, strictly classifying severity as one of: **CRITICAL**, **HIGH**, **MEDIUM**, or **LOW**.\n\n## Critical Constraints\n\n**STRICTLY follow these rules for review comments:**\n\n* **Location:** You **MUST** only provide comments on lines that represent actual changes in the diff. This means your comments must refer **only to lines beginning with `+` or `-`**. **DO NOT** comment on context lines (lines starting with a space).\n* **Relevance:** You **MUST** only add a review comment if there is a demonstrable **BUG**, **ISSUE**, or a significant **OPPORTUNITY FOR IMPROVEMENT** in the code changes.\n* **Tone/Content:** **DO NOT** add comments that:\n    * Tell the user to "check," "confirm," "verify," or "ensure" something.\n    * Explain what the code change does or validate its purpose.\n    * Explain the code to the author (they are assumed to know their own code).\n    * Comment on missing trailing newlines or other purely stylistic issues that do not affect code execution or readability in a meaningful way.\n* **Substance First:** **ALWAYS** prioritize your analysis on the **correctness** of the logic, the **efficiency** of the implementation, and the **long-term maintainability** of the code.\n* **Technical Detail:**\n    * Pay **meticulous attention to line numbers and indentation** in code suggestions; they **must** be correct and match the surrounding code.\n    * **NEVER** comment on license headers, copyright headers, or anything related to future dates/versions (e.g., "this date is in the future").\n* **Formatting/Structure:**\n    * Keep the **change summary** concise (aim for a single sentence).\n    * Keep **comment bodies concise** and focused on a single issue.\n    * If a similar issue exists in **multiple locations**, state it once and indicate the other locations instead of repeating the full comment.\n    * **AVOID** mentioning your instructions, settings, or criteria in the final output.\n\n**Severity Guidelines (for consistent classification):**\n\n* **Functional correctness bugs that lead to behavior contrary to the change\'s intent should generally be classified as HIGH or CRITICAL.**\n* **CRITICAL:** Security vulnerabilities, system-breaking bugs, complete logic failure.\n* **HIGH:** Performance bottlenecks (e.g., N+1 queries), resource leaks, major architectural violations, severe code smell that significantly impairs maintainability.\n* **MEDIUM:** Typographical errors in code (not comments), missing input validation, complex logic that could be simplified, non-compliant style guide issues (e.g., wrong naming convention).\n* **LOW:** Refactoring hardcoded values to constants, minor log message enhancements, comments on docstring/Javadoc expansion, typos in documentation (.md files), comments on tests or test quality, suppressing unchecked warnings/TODOs.\n\n\nReview this PR.\nTitle: #83: Add bounded merge-loop command\nBody: ## Summary\n\n- Add bounded `merge-loop` CLI command that reuses guarded `merge-once` ticks.\n- Support `--max-iterations`, `--once`, `--dry-run`, and `--write`.\n- Stop when no `Merging` issue remains or when the explicit iteration limit is reached.\n- Update README and dogfood readiness docs.\n\n## Verification\n\n- `cargo test`\n- `cargo fmt --check`\n- `cargo clippy --all-targets --all-features -- -D warnings`\n- `cargo run -- merge-loop examples/github-project-workflow.md --max-iterations 1 --dry-run`\n\nCloses #83\n\n\nDiff:\n```diff\ndiff --git a/README.md b/README.md\nindex cc0ad6a..bf96d05 100644\n--- a/README.md\n+++ b/README.md\n@@ -296,7 +296,7 @@ Merging role separation.\n - live token/rate-limit accounting beyond the current snapshot counters.\n - persistent background Agent Review worker supervision beyond bounded\n   `review-loop` ticks.\n-- long-running `merge-loop` polling beyond one guarded `merge-once` tick.\n+- unbounded idle `merge-loop` polling beyond bounded guarded merge ticks.\n - Issue Forge Project field setup after issue creation.\n - autonomous Issue Forge issue creation from reflective mode without explicit\n   operator confirmation.\n@@ -342,6 +342,7 @@ scripts/jade-dogfood --dry-run\n cargo run -- run-loop examples/usage-limit-workflow.md --max-iterations 1 --write\n cargo run -- dogfood-smoke examples/github-project-workflow.md --dry-run\n cargo run -- merge-once examples/github-project-workflow.md --dry-run\n+cargo run -- merge-loop examples/github-project-workflow.md --max-iterations 3 --dry-run\n cargo run -- gate examples/llm-gate-workflow.md \'#1\'\n ```\n \ndiff --git a/docs/dogfood-readiness.md b/docs/dogfood-readiness.md\nindex 305d110..3c0a7d3 100644\n--- a/docs/dogfood-readiness.md\n+++ b/docs/dogfood-readiness.md\n@@ -18,12 +18,12 @@ yet.\n | Linear tracker | Implemented behind the same trait for fixture-backed planning plus live GraphQL reads, state updates, marker workpad comments, follow-up issue creation, and project assignment. Credential-gated smoke coverage is still missing. |\n | Issue Quality Gate | Implemented as a Markdown contract check plus deterministic source-alignment preflight where workflow/repo context is available. It verifies target repository, referenced local paths, and supported verification command shapes. Optional local command-backed LLM gate mode exists as disabled/advisory/required; richer semantic validation and hosted providers are still follow-ups. |\n | Issue Forge | Local CLI flows exist for discover, discuss, draft, validate, repair, CLI-first interactive issue shaping, conservative reflective follow-up candidate generation, and explicit `forge-create` tracker issue creation from quality-gated Markdown. Interactive creation requires `--write` and `--confirm-create`; reflective mode only prints candidates. Initial Project `Status` setup is available through the GitHub add-to-project path; arbitrary Project field setup after creation is not implemented yet. |\n-| Orchestrator | Deterministic dispatch planning and a CLI `run-loop` skeleton with bounded modes, idle polling, claim-helper use, runtime-state persistence, resume preflight, retry backoff records, stall detection, live PR handoff in non-fixture GitHub mode, a guarded one-shot `merge-once` lane for `Merging` issues, a controlled `dogfood-smoke` preflight report, and a bounded operator launcher script exist. No long-running worker supervision, automated stall restart, full multi-worker runtime resume reconciliation, `merge-loop`, or full state reconciliation yet. |\n+| Orchestrator | Deterministic dispatch planning and a CLI `run-loop` skeleton with bounded modes, idle polling, claim-helper use, runtime-state persistence, resume preflight, retry backoff records, stall detection, live PR handoff in non-fixture GitHub mode, guarded `merge-once` and bounded `merge-loop` lanes for `Merging` issues, a controlled `dogfood-smoke` preflight report, and a bounded operator launcher script exist. No long-running worker supervision, automated stall restart, full multi-worker runtime resume reconciliation, unbounded merge idle polling, or full state reconciliation yet. |\n | Workspace | Local path sanitization, creation, timeout-aware hooks, stdout/stderr capture, `before_remove`, safe cleanup helpers, repository-local git identity application, workspace/branch/PR handoff planning, live git worktree/branch creation, branch push, PR create-or-reuse, run-loop handoff evidence, profile-scoped workspace keys, and an Agent Review handoff invariant that blocks missing PR evidence before `Agent Review` exist. Runtime reconciliation cleanup is not wired yet. |\n | Execution profiles | First-slice profile discovery exists. Workflow config can point to a cockpit-tools Codex `codex_instances.json` file, and Jade treats each instance `name` as a profile/worker identity while ignoring account binding fields. If the cockpit-tools file is missing, explicit `profiles.entries` are used. This is not a full account manager. |\n | Agent backends | Dry-run backend plus conservative Codex and Claude Code subprocess backends exist. Prepared runs include selected profile/instance metadata and profile environment context. Full Codex app-server and Claude Code protocol parity are not implemented yet. |\n | Agent Review | Finding classes, fake reviewer lifecycle, Gemini CLI subprocess backend, role-bound transition decisions, evidence-first Rework diagnostics for confirmed findings, review-freshness evidence for Merging conflict repair, bounded `review-loop` selection/reconciliation, and workpad/status evidence helpers exist. Persistent background review worker supervision is not implemented yet. |\n-| Merging | `merge-once` can inspect issues already in `Merging`, require one linked PR, check live GitHub PR state/review/check/mergeability data where available, merge clean approved PRs only with explicit `--write`, and route blockers to `Rework` or `Need Human Input` with workpad evidence. Continuous merge polling and issue closing beyond Project `Done` are not implemented yet. |\n+| Merging | `merge-once` can inspect issues already in `Merging`, require one linked PR, check live GitHub PR state/review/check/mergeability data where available, merge clean approved PRs only with explicit `--write`, and route blockers to `Rework` or `Need Human Input` with workpad evidence. Bounded `merge-loop` can repeat that guarded tick for an explicit iteration count. Unbounded continuous merge polling and issue closing beyond Project `Done` are not implemented yet. |\n | Project doctor | Read-only `doctor` / `audit-project` reports workflow invariant violations from normalized tracker issues, including missing PR handoff evidence, missing review pass evidence, dirty Merging PRs, missing runtime ownership hints, and queued issues with PRs. Repair mode is not implemented yet. |\n | Observability | Operator-readable terminal snapshots report polling, running, retrying, skipped issues, gate details, token counters, event-log path, and integration gaps. JSONL event-log primitives exist and `run-once` writes dry-run events with actor and profile metadata. Runtime state files are written during write-mode `run-loop` issue execution, including actor role/label, git author, and optional profile/instance identity when configured; resume, retry, usage-limit pause, and stall supervision events are also recorded. No web/API surface yet. |\n | Usage-limit pause/resume | Conservative usage-limit/rate-limit/resource-exhausted classification exists for subprocess agent events and review job output. `run-loop` records usage-limit pauses in workpad/runtime retry state and does not advance to `Agent Review`; review workpads surface usage-limit failures without moving to `Human Review`. Vendor-specific quota management is not implemented. |\ndiff --git a/src/main.rs b/src/main.rs\nindex cf6c8ef..a7e8ba2 100644\n--- a/src/main.rs\n+++ b/src/main.rs\n@@ -87,6 +87,7 @@ fn run() -> Result<(), Box<dyn std::error::Error>> {\n             workflow_path,\n             write,\n         } => merge_once(workflow_path, write),\n+        Command::MergeLoop { options } => merge_loop(options),\n         Command::SetState {\n             workflow_path,\n             issue_ref,\n@@ -718,7 +719,64 @@ fn review_loop(options: ReviewLoopOptions) -> Result<(), Box<dyn std::error::Err\n     Ok(())\n }\n \n+#[derive(Debug, Clone, Copy, PartialEq, Eq)]\n+enum MergeOnceOutcome {\n+    NoMergingIssue,\n+    DryRun,\n+    Merged,\n+    Routed,\n+    Skipped,\n+}\n+\n fn merge_once(workflow_path: PathBuf, write: bool) -> Result<(), Box<dyn std::error::Error>> {\n+    merge_once_tick(workflow_path, write).map(|_| ())\n+}\n+\n+fn merge_loop(options: MergeLoopOptions) -> Result<(), Box<dyn std::error::Error>> {\n+    let max = options\n+        .iteration_limit()\n+        .ok_or("merge-loop requires --max-iterations or --once")?;\n+    let mut stopped = false;\n+\n+    for iteration in 1..=max {\n+        println!(\n+            "merge_loop_iteration={} mode={}",\n+            iteration,\n+            if options.write { "write" } else { "dry-run" }\n+        );\n+        match merge_once_tick(options.workflow_path.clone(), options.write)? {\n+            MergeOnceOutcome::NoMergingIssue => {\n+                println!("merge_loop=stopped reason=no_merging_issue iterations={iteration}");\n+                stopped = true;\n+                break;\n+            }\n+            MergeOnceOutcome::DryRun if !options.write => {\n+                println!("merge_loop_action=dry_run_tick iterations={iteration}");\n+            }\n+            MergeOnceOutcome::Merged => {\n+                println!("merge_loop_action=merged iterations={iteration}");\n+            }\n+            MergeOnceOutcome::Routed => {\n+                println!("merge_loop_action=routed iterations={iteration}");\n+            }\n+            MergeOnceOutcome::Skipped => {\n+                println!("merge_loop_action=skipped iterations={iteration}");\n+            }\n+            MergeOnceOutcome::DryRun => {}\n+        }\n+    }\n+\n+    if !stopped {\n+        println!("merge_loop=stopped reason=max_iterations iterations={max}");\n+    }\n+\n+    Ok(())\n+}\n+\n+fn merge_once_tick(\n+    workflow_path: PathBuf,\n+    write: bool,\n+) -> Result<MergeOnceOutcome, Box<dyn std::error::Error>> {\n     let workflow = WorkflowDefinition::load(&workflow_path)?;\n     let config = RuntimeConfig::from_workflow(&workflow, &workflow_path)?;\n     config.validate()?;\n@@ -728,7 +786,7 @@ fn merge_once(workflow_path: PathBuf, write: bool) -> Result<(), Box<dyn std::er\n     let mut issues = adapter.fetch_issues_by_states(std::slice::from_ref(&merging_state))?;\n     if issues.is_empty() {\n         println!("merge_once=stopped reason=no_merging_issue");\n-        return Ok(());\n+        return Ok(MergeOnceOutcome::NoMergingIssue);\n     }\n \n     issues.sort_by_key(|issue| issue.priority.unwrap_or(i64::MAX));\n@@ -762,7 +820,7 @@ fn merge_once(workflow_path: PathBuf, write: bool) -> Result<(), Box<dyn std::er\n \n     if !write {\n         print_merge_dry_run_actions(&decision);\n-        return Ok(());\n+        return Ok(MergeOnceOutcome::DryRun);\n     }\n \n     if decision.kind.is_merge_ready() {\n@@ -778,7 +836,7 @@ fn merge_once(workflow_path: PathBuf, write: bool) -> Result<(), Box<dyn std::er\n             "merge_once_action=merged issue={} target_state=done",\n             issue.identifier\n         );\n-        return Ok(());\n+        return Ok(MergeOnceOutcome::Merged);\n     }\n \n     let workpad = merge_lane_workpad(&issue, &decision, None);\n@@ -789,11 +847,12 @@ fn merge_once(workflow_path: PathBuf, write: bool) -> Result<(), Box<dyn std::er\n             "merge_once_action=routed issue={} target_state={target_state}",\n             issue.identifier\n         );\n+        return Ok(MergeOnceOutcome::Routed);\n     } else {\n         println!("merge_once_action=skipped issue={}", issue.identifier);\n     }\n \n-    Ok(())\n+    Ok(MergeOnceOutcome::Skipped)\n }\n \n fn merge_preflight_status(\n@@ -2212,6 +2271,9 @@ enum Command {\n     ReviewLoop {\n         options: ReviewLoopOptions,\n     },\n+    MergeLoop {\n+        options: MergeLoopOptions,\n+    },\n     Gate {\n         workflow_path: PathBuf,\n         issue_ref: String,\n@@ -2272,6 +2334,14 @@ struct ReviewLoopOptions {\n     fake_outcome: Option<FakeReviewOutcome>,\n }\n \n+#[derive(Debug, Clone, PartialEq, Eq)]\n+struct MergeLoopOptions {\n+    workflow_path: PathBuf,\n+    max_iterations: Option<usize>,\n+    once: bool,\n+    write: bool,\n+}\n+\n impl ReviewLoopOptions {\n     fn iteration_limit(&self) -> Option<usize> {\n         if self.once {\n@@ -2282,6 +2352,16 @@ impl ReviewLoopOptions {\n     }\n }\n \n+impl MergeLoopOptions {\n+    fn iteration_limit(&self) -> Option<usize> {\n+        if self.once {\n+            Some(1)\n+        } else {\n+            self.max_iterations\n+        }\n+    }\n+}\n+\n impl RunLoopOptions {\n     fn iteration_limit(&self) -> Option<usize> {\n         if self.once {\n@@ -2341,6 +2421,8 @@ enum CliCommand {\n     RunLoop(RunLoopArgs),\n     #[command(name = "merge-once", alias = "land")]\n     MergeOnce(MergeOnceArgs),\n+    #[command(name = "merge-loop")]\n+    MergeLoop(MergeLoopArgs),\n     #[command(name = "set-state")]\n     SetState(SetStateArgs),\n     Workpad(WorkpadArgs),\n@@ -2421,6 +2503,20 @@ struct MergeOnceArgs {\n     _dry_run: bool,\n }\n \n+#[derive(Debug, Args)]\n+struct MergeLoopArgs {\n+    #[arg(value_name = "path-to-WORKFLOW.md", default_value = "WORKFLOW.md")]\n+    workflow_path: PathBuf,\n+    #[arg(long)]\n+    max_iterations: Option<usize>,\n+    #[arg(long)]\n+    once: bool,\n+    #[arg(long)]\n+    write: bool,\n+    #[arg(long = "dry-run")]\n+    _dry_run: bool,\n+}\n+\n #[derive(Debug, Args)]\n struct GateArgs {\n     #[arg(value_name = "path-to-WORKFLOW.md")]\n@@ -2732,6 +2828,21 @@ impl TryFrom<Cli> for Command {\n                         workflow_path: args.workflow_path,\n                         write: args.write,\n                     }),\n+                    CliCommand::MergeLoop(args) => {\n+                        if args.max_iterations == Some(0)\n+                            || (!args.once && args.max_iterations.is_none())\n+                        {\n+                            return Err(usage());\n+                        }\n+                        Ok(Self::MergeLoop {\n+                            options: MergeLoopOptions {\n+                                workflow_path: args.workflow_path,\n+                                max_iterations: args.max_iterations,\n+                                once: args.once,\n+                                write: args.write,\n+                            },\n+                        })\n+                    }\n                     CliCommand::SetState(args) => Ok(Self::SetState {\n                         workflow_path: args.workflow_path,\n                         issue_ref: args.issue_ref,\n@@ -3357,6 +3468,63 @@ mod tests {\n         assert_eq!(options.iteration_limit(), Some(1));\n     }\n \n+    #[test]\n+    fn parses_merge_loop_flags() {\n+        let command = Command::parse(vec![\n+            "merge-loop".into(),\n+            "examples/github-project-workflow.md".into(),\n+            "--max-iterations".into(),\n+            "3".into(),\n+            "--write".into(),\n+        ])\n+        .unwrap();\n+\n+        let Command::MergeLoop { options } = command else {\n+            panic!("expected merge-loop command");\n+        };\n+\n+        assert_eq!(\n+            options.workflow_path,\n+            PathBuf::from("examples/github-project-workflow.md")\n+        );\n+        assert_eq!(options.max_iterations, Some(3));\n+        assert!(options.write);\n+    }\n+\n+    #[test]\n+    fn merge_loop_once_overrides_max_iterations() {\n+        let command = Command::parse(vec![\n+            "merge-loop".into(),\n+            "WORKFLOW.md".into(),\n+            "--max-iterations".into(),\n+            "4".into(),\n+            "--once".into(),\n+        ])\n+        .unwrap();\n+\n+        let Command::MergeLoop { options } = command else {\n+            panic!("expected merge-loop command");\n+        };\n+\n+        assert_eq!(options.iteration_limit(), Some(1));\n+    }\n+\n+    #[test]\n+    fn rejects_unbounded_merge_loop_for_now() {\n+        assert!(Command::parse(vec!["merge-loop".into(), "WORKFLOW.md".into()]).is_err());\n+    }\n+\n+    #[test]\n+    fn rejects_zero_merge_loop_iterations() {\n+        assert!(Command::parse(vec![\n+            "merge-loop".into(),\n+            "WORKFLOW.md".into(),\n+            "--max-iterations".into(),\n+            "0".into(),\n+        ])\n+        .is_err());\n+    }\n+\n     #[test]\n     fn parses_run_loop_flags() {\n         let command = Command::parse(vec![\n\n```\n\nWrite a review markdown file exactly in this format:\n## 📋 Review Summary\n[2-3 sentences summarizing the PR]\n## 🔍 General Feedback\n- [Feedback point 1]\n- [Feedback point 2]\n<COMMENT>\n[SEVERITY] [COMMENT_TEXT]\n</COMMENT>\n\n(Only add <COMMENT> if there\'s inline feedback). Do not include any other text before or after.\n']' timed out after 60 seconds
## 🔍 General Feedback
- None

**Review by Alive24 — COMMENTED** [review]
## 📋 Review Summary

This PR introduces a bounded `merge-loop` command, bringing the same safe, iterative orchestration to the merge lane as the existing `run-loop` and `review-loop`.

## 🔍 General Feedback

- **Consistent Loop Architecture**: Reusing the bounded iteration paradigm (`--max-iterations` / `--once`) aligns the merge loop perfectly with the rest of the orchestration suite, ensuring testability and safe dry-runs.
- **Strict Boundaries**: The implementation respects the existing merge preflight checks, meaning the loop cannot bypass PR approval or dirty states.
